realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1512,17 +1512,6 @@ JNIEXPORT jstring JNICALL Java_io_realm_internal_Table_nativeToJson(
     return NULL;
 }
 
-JNIEXPORT jboolean JNICALL Java_io_realm_internal_Table_nativeEquals(
```

Comment:
If this function is to compare pointers only, it would be fine to do so in Java as we have native pointers there.